### ISSUE TITLE
ABNF recipient list production #element does not allow null element prior to first element

### ISSUE THREAD

**Issue by reschke** [httpwg/http-core#164]
See <https://www.rfc-editor.org/errata/eid5257>.

Mailing list discussion starts here: <https://lists.w3.org/Archives/Public/ietf-http-wg/2018JanMar/0131.html>

**Comment by reschke** [comment]
See also #14

**Comment by reschke** [comment]
A simple fix for this would be to simplify the list expansion to just map `\#element`, and to put all other constraints in prose (the current approach doesn't work anyway due to the rules for empty list elements which are not counted).

**Comment by reschke** [comment]
Proposal for recipient list rule mapping:

~~~
#element => *( OWS "," ) *( [ OWS element ] OWS "," )
~~~

More eyes requested :-) @royfielding @mnot

**Comment by reschke** [comment]
The current proposal is incorrect as it does not allow a singleton element.

**Comment by reschke** [comment]
Maybe:

~~~
#element => *( OWS "," ) [ element ] *( OWS "," [ OWS element ] ) *( OWS "," )
~~~

**Comment by reschke** [comment]
PR updated

**Comment by royfielding** [comment]
And maybe

    1#element => element *( OWS "," OWS [ element ] )

right?

**Comment by annevk** [comment]
@royfielding that'd outlaw `Multiple-Element-Header: , somevalue`, which seems wrongish?

**Comment by royfielding** [comment]
Closed by commit e475f36 for the recipient case. Note that this may become obsolete if we switch to only providing ABNF for what is good to send.
